lzfse/src/lib.rs

```diff
@@ -32,62 +32,69 @@
 //! assert_eq!(input[..], uncompressed[..bytes_in]);
 //! ````
 
-extern crate libc;
-extern crate lzfse_sys as ffi;
-
 use libc::size_t;
+use lzfse_sys as ffi;
+use std::ptr;
 
-/// This type represents all possible errors that can occur when decompressing data.
+/// This type represents all possible errors that can occur when
+/// (de)compressing data.
 #[derive(PartialEq, Debug)]
 pub enum Error {
-    /// The buffer was not large enough for the decompressed data.
     BufferTooSmall,
-    /// Decompression failed because the input was invalid.
     CompressFailed,
 }
 
-/// Compress input into byte array
+/// Compress input into byte array.
+///
+/// # Errors
+///
+/// - Returns `CompressFailed` if compression failed for any reason, most
+///   likely because the output buffer is too small.
 pub fn encode_buffer(input: &[u8], output: &mut [u8]) -> Result<usize, Error> {
     let out_size = unsafe {
         ffi::lzfse_encode_buffer(
             output.as_ptr() as *mut _,
             output.len() as size_t,
             input.as_ptr() as *const _,
             input.len() as size_t,
-            0 as *mut _,
+            ptr::null_mut(),
         ) as usize
     };
 
-    if out_size == 0 {
-        Err(Error::CompressFailed)
-    } else {
-        Ok(out_size)
+    match out_size {
+        0 => Err(Error::CompressFailed),
+        n => Ok(n),
     }
 }
 
-/// Decompress input into byte array
+/// Decompress input into byte array,
+///
+/// # Errors
+///
+/// - Returns `BufferTooSmall` if the output buffer is too small to fit all
+///   decompressed data.
+/// - Returns `CompressFailed` if the decompression failed for any reason,
+///   most likely because of invalid input data.
 pub fn decode_buffer(input: &[u8], output: &mut [u8]) -> Result<usize, Error> {
     let out_size = unsafe {
         ffi::lzfse_decode_buffer(
             output.as_ptr() as *mut _,
             output.len() as size_t,
             input.as_ptr() as *const _,
             input.len() as size_t,
-            0 as *mut _,
+            ptr::null_mut(),
         ) as usize
     };
 
-    if out_size == output.len() {
-        Err(Error::BufferTooSmall)
-    } else {
-        Ok(out_size)
+    match out_size {
+        0 => Err(Error::CompressFailed),
```

Comment:
If you compress an empty slice, when you decompress it, I believe you'll get back a zero correctly..